src/main/java/htsjdk/utils/ValidationUtils.java

```diff
@@ -0,0 +1,150 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package htsjdk.utils;
+
+import java.util.Collection;
+import java.util.function.Supplier;
+
+/**
+ * Simple functions that streamline the checking of values.
+ */
+public class ValidationUtils {
+    /**
+     * Checks that an Object {@code object} is not null and returns the same object or throws an {@link IllegalArgumentException}
+     * @param object any Object
+     * @return the same object
+     * @throws IllegalArgumentException if a {@code o == null}
+     */
+    public static <T> T nonNull(final T object) {
+        return ValidationUtils.nonNull(object, "Null object is not allowed here.");
+    }
+
+    /**
+     * Checks that an {@link Object} is not {@code null} and returns the same object or throws an {@link IllegalArgumentException}
+     * @param object any Object
+     * @param message the text message that would be passed to the exception thrown when {@code o == null}.
+     * @return the same object
+     * @throws IllegalArgumentException if a {@code o == null}
+     */
+    public static <T> T nonNull(final T object, final String message) {
+        if (object == null) {
+            throw new IllegalArgumentException(message);
+        }
+        return object;
+    }
+
+    /**
+     * Checks that an {@link Object} is not {@code null} and returns the same object or throws an {@link IllegalArgumentException}
+     * @param object any Object
+     * @param message the text message that would be passed to the exception thrown when {@code o == null}.
+     * @return the same object
+     * @throws IllegalArgumentException if a {@code o == null}
+     */
+    public static <T> T nonNull(final T object, final Supplier<String> message) {
+        if (object == null) {
+            throw new IllegalArgumentException(message.get());
+        }
+        return object;
+    }
+
+
+    /**
+     * Checks that a {@link Collection} is not {@code null} and that it is not empty.
+     * If it's non-null and non-empty it returns the input, otherwise it throws an {@link IllegalArgumentException}
+     * @param collection any Collection
+     * @param message a message to include in the output
+     * @return the original collection
+     * @throws IllegalArgumentException if collection is null or empty
+     */
+    public static <I, T extends Collection<I>> T nonEmpty(T collection, String message){
```

Comment:
add space before `{`. many missing spaces in this file, would be easier to re-format in the IDE to fix them all